src/vma/dev/qp_mgr.cpp

```diff
@@ -123,10 +124,53 @@ qp_mgr::~qp_mgr()
 	munmap(m_rq_wqe_idx_to_wrid, m_rx_num_wr * sizeof(*m_rq_wqe_idx_to_wrid));
 	munmap(m_sq_wqe_idx_to_wrid, m_tx_num_wr * sizeof(*m_sq_wqe_idx_to_wrid));
 #endif // DEFINED_VMAPOLL
+
+#ifndef DEFINED_VMAPOLL
+	delete[] m_rq_wqe_idx_to_wrid;
+#else
+	free(m_rq_wqe_idx_to_wrid);
```

Comment:
If `defined(DEFINE_VMAPOLL)` then `m_rq_wqe_idx_to_wrid` is `munmap`ed. No need to free here.